File: apps/flutter_app/lib/src/core/demo_data.dart
import 'models.dart';

const markets = <Market>[
  Market(
    id: 'btc-120k',
    title: 'Will BTC exceed \$120k before Dec 31 2026?',
    category: 'Crypto',
    yesProbability: 0.74,
    aiConfidence: 0.91,
    volume: 842300,
    liquidity: 320000,
    points: [0.65, 0.68, 0.71, 0.74],
  ),
  Market(
    id: 'hashkey-tvl',
    title: 'Will HashKey Chain TVL exceed \$50M by Q3?',
    category: 'Ecosystem',
    yesProbability: 0.58,
    aiConfidence: 0.79,
    volume: 265800,
    liquidity: 180000,
    points: [0.49, 0.52, 0.55, 0.58],
  ),
  Market(
    id: 'eth-etf',
    title: 'Will ETH ETF volume double by year end?',
    category: 'Macro',
    yesProbability: 0.63,
    aiConfidence: 0.84,
    volume: 410500,
    liquidity: 245000,
    points: [0.55, 0.59, 0.61, 0.63],
  ),
];

const agents = <AgentCardModel>[
  AgentCardModel(
    name: 'Liquidity Agent',
    status: 'Intervening',
    summary: 'Injecting depth into BTC macro markets before volatility windows.',
    pnl: 18420,
  ),
  AgentCardModel(
    name: 'Arbitrage Agent',
    status: 'Active',
    summary: 'Capturing spread dislocations between venue clusters.',
    pnl: 9260,
  ),
  AgentCardModel(
    name: 'Sentinel Agent',
    status: 'Watching',
    summary: 'Monitoring whale wallet clusters and abnormal coordination.',
    pnl: 0,
  ),
];

File: apps/backend/app/services/demo_data.py
from datetime import datetime


def demo_markets():
    return [
        {
            "id": 1,
            "slug": "btc-120k-2026",
            "title": "Will BTC exceed $120k before Dec 31 2026?",
            "description": "Resolves YES if BTC exceeds 120,000 USD on tracked exchanges before year end 2026.",
            "category": "Crypto",
            "oracle_source": "HashKey oracle mesh",
            "expiry_at": datetime.fromisoformat("2026-12-31T23:59:00"),
            "yes_probability": 0.74,
            "no_probability": 0.26,
            "ai_confidence": 0.91,
            "volume": 842300.0,
            "liquidity": 320000.0,
            "resolved": False,
            "outcome": "PENDING",
        },
        {
            "id": 2,
            "slug": "hashkey-tvl-q3",
            "title": "Will HashKey Chain TVL exceed $50M by Q3?",
            "description": "Resolves YES if HashKey Chain TVL exceeds 50M USD by Q3 close.",
            "category": "Ecosystem",
            "oracle_source": "TVL oracle bridge",
            "expiry_at": datetime.fromisoformat("2026-09-30T23:59:00"),
            "yes_probability": 0.58,
            "no_probability": 0.42,
            "ai_confidence": 0.79,
            "volume": 265800.0,
            "liquidity": 180000.0,
            "resolved": False,
            "outcome": "PENDING",
        },
        {
            "id": 3,
            "slug": "eth-etf-volume",
            "title": "Will ETH ETF volume double by year end?",
            "description": "Resolves YES if regulated ETH ETF volume doubles against Jan 1 baseline.",
            "category": "Macro",
            "oracle_source": "Institutional ETF reporting oracle",
            "expiry_at": datetime.fromisoformat("2026-12-31T23:59:00"),
            "yes_probability": 0.63,
            "no_probability": 0.37,
            "ai_confidence": 0.84,
            "volume": 410500.0,
            "liquidity": 245000.0,
            "resolved": False,
            "outcome": "PENDING",
        },
    ]


def demo_positions():
    return [
        {
            "market_title": "Will BTC exceed $120k before Dec 31 2026?",
            "side": "YES",
            "size": 4200.0,
            "avg_price": 0.61,
            "mark_price": 0.74,
            "pnl": 546.0,
        },
        {
            "market_title": "Will HashKey Chain TVL exceed $50M by Q3?",
            "side": "YES",
            "size": 1900.0,
            "avg_price": 0.52,
            "mark_price": 0.58,
            "pnl": 114.0,
        },
    ]


def demo_agents():
    return [
        {
            "agent": "Liquidity Agent",
            "status": "INTERVENING",
            "pnl": 18420,
            "interventions": 21,
            "summary": "Injecting depth into high-volatility markets ahead of macro catalysts.",
        },
        {
            "agent": "Sentinel Agent",
            "status": "WATCHING",
            "pnl": 0,
            "interventions": 7,
            "summary": "Monitoring suspicious wallet clusters and abnormal volume spikes.",
        },
    ]
